generator/svelte/src/lib/utils.ts

```diff
@@ -40,15 +41,82 @@ export const isOlderThanAWeek = (isoDate: string) => {
 	return dayjs.utc(isoDate).isBefore(dayjs.utc().subtract(7, 'day'), 'day');
 };
 
+export const getCurrentLocalDateTime = () => {
+	return dayjs.utc().local();
+};
+
 /** Data transformation */
 
-export const extractPrayersFromMasjids = (masjids: [string, IMasjid][]) => {
-	if (!masjids.length) return [];
-	const masjid = masjids[0][1];
-	if (!masjid) return [];
+export const getPrayers = () => {
+	const prayers: Partial<TPrayer>[] = PRAYER_NAMES.map((prayer) => ({ name: prayer }));
+	const tPrayers: TPrayer[] = [];
+
+	for (let i = 0; i < prayers.length; i++) {
+		const prayer = prayers[i];
+		const next = prayers[i + 1] || prayers[0];
+		prayer.next = next as TPrayer;
+		tPrayers.push(prayer as TPrayer);
+	}
+
+	return tPrayers;
+};
+
+export const getNextPrayerForMasjid = (masjid: { iqamas: Record<string, { time: string }> }) => {
+	const currentTime = getCurrentLocalDateTime();
+	const prayers = getPrayers();
+
+	const nextPrayer = prayers.find((prayer) => {
+		const prayerTime = masjid.iqamas[prayer.name]?.time.toUpperCase();
+		const prayerDateTime = dayjs(prayerTime, 'h:mm A').local();
+		if (!prayerTime) return false;
+		return prayerDateTime.isAfter(currentTime);
+	});
+
+	if (!nextPrayer) {
+		throw new Error('Something wrong, No current prayer found');
```

Comment:
How does this exception affect the view? We need a safe fallback.

Reply:
I thought it should never happen, what could we fallback to?

Reply:
@hammady Added fallback to first prayer with an error log